openedx/core/djangoapps/user_authn/tasks.py

```diff
@@ -13,12 +15,51 @@
 from edx_ace.errors import RecoverableChannelDeliveryError
 from edx_ace.message import Message
 from edx_django_utils.monitoring import set_code_owner_attribute
+from rest_framework.status import HTTP_408_REQUEST_TIMEOUT
 
+from common.djangoapps.track import segment
+from openedx.core.djangoapps.password_policy.hibp import PwnedPasswordsAPI
 from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_authn.config.waffle import ENABLE_PWNED_PASSWORD_API
 from openedx.core.lib.celery.task_utils import emulate_http_request
 
 log = logging.getLogger('edx.celery.task')
 
+def get_pwned_properties(pwned_response, password):
+    """
+    Derive different pwned parameters for analytics
+    """
+    properties = {}
+    if pwned_response == HTTP_408_REQUEST_TIMEOUT:
+        properties['vulnerability'] = 'unknown'
+        pwned_count = 0
+    else:
+        pwned_count = pwned_response.get(password, 0)
+        properties['vulnerability'] = 'yes' if pwned_count > 0 else 'no'
+
+    if pwned_count > 0:
+        properties['frequency'] = math.ceil(math.log10(pwned_count))
+
+    return properties
+
+@shared_task
+def check_pwned_password(user_id, password, internal_user=False):
```

Comment:
Since this sending a track event as well, can we rename it accordingly? e.g. `check_pwned_password_and_send_track_event`